spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -208,6 +239,12 @@ public MessageGroup addMessageToGroup(Object groupId, Message<?> message) {
 				if (group == null) {
 					group = new SimpleMessageGroup(groupId);
 					this.groupIdToMessageGroup.putIfAbsent(groupId, group);
+					this.groupToUpperBound.putIfAbsent(groupId, new UpperBound(this.groupCapacity));
+				}
+				if (!this.groupToUpperBound.get(groupId).tryAcquire(this.upperBoundTimeout)) {
```

Comment:
We either need a new catch for `InterruptedException` here, or at least we need to fix the message in the `MessagingException` in the outer catch since the interrupt might be here not on the lock acquisition.